src/mon/MDSMonitor.cc

```diff
@@ -988,12 +988,16 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
       ostringstream ess;
       mds_gid_t gid = gid_from_arg(fsmap, id, ess);
       if (gid == MDS_GID_NONE) {
-	// the mds doesn't exist, but no file systems are unhappy, so losing it
-	// can't have any effect.
-	continue;
+        invalid_ids.push_back(std::string(id));
+        continue;
       }
       stopping.insert(gid);
     }
+    if (!invalid_ids.empty()) {
+      r = -EINVAL;
```

Comment:
If any, then the appropriate return code for this kind of failure should be `ENOENT`.

`EINVAL` should be returned for a badly formatted request, or a request that is inconsistent. Like, if the name wasn't parseable, or if multiple conflicting commands were attempted together.

However, we must double-check with the ticket authors whether it's OK to change the status output of this command. The thing is, while it does make sense to validate input, existing scripts might not expect this command to fail in this scenario, and could misinterpret the signal.

`ok-to-stop` should be used to determine whether any of the ids provided refers to a service that shouldn't be stopped. If that is not the case (including when id is invalid), the caller expects a successful result code. The command to stop an MDS, if the caller gets to that, will check for a valid mds id, which is expected.

I would suggest keeping this PR to add the detailed output on the stderr about the issues we encounter when trying to parse the provided ids, but not change the logic that generates the response code.

@lordcirth, as the creator of the issue, wdyt? @gregsfortytwo FYI. 
ticket: https://tracker.ceph.com/issues/42597

Reply:
I think it should be fine to return an error. If that causes existing scripts to fail, well, I'd say its for the better :)

I don't think we enforce _strict_ backward compatibility for things like this.

Reply:
> @lordcirth, as the creator of the issue, wdyt?

In the case that led me to report it, I was running it manually, so stderr would work; but if I were writing a script I would expect it to return non-zero if the name didn't exist. ENOENT does seem to be the correct code for that.

Reply:
I discussed this with @gregsfortytwo a bit - erroring out is fine since we do not have strict rules for such things.